### PR TITLE
Replace Elasticlunr with ElasticSearch

### PR DESCRIPTION
**Pull request by Hackin7**
image

Replaced the search results from elasticlunr (which doesn't work on my system for some reason) with ElasticSearch (which does work). API Key used is mine

XMLHttpRequest used to retrieve data from the endpoint, no need for external libraries.

Code will be cleaned up soon

### CODE REVIEW

**Comment by Hackin7** [comment]
For now, this seems like a decent choice, especially since mentorship is already using it, and the code to interface with ElasticSearch is so minimal

However, if it is needed to interface with the endpoint in more ways than just search, this may not scale 

May need consider other alternatives.

API used is this (happens to be how mentorship does it too)
https://www.elastic.co/guide/en/app-search/current/search.html

File: assets/js/main.js [hunk 1]
```diff
@@ -266,95 +265,43 @@ function modal() {
...
+
+    var oReq = new XMLHttpRequest();
+    var payload = { query: query };
+    oReq.open(
+        'POST',
+        base_url + '/api/as/v1/engines/' + engine + '/search.json'
+    );
+    oReq.addEventListener('load', callback);
+    oReq.setRequestHeader('Content-Type', 'application/json');
+    oReq.setRequestHeader('Authorization', 'Bearer ' + search_only_key);
```
Comment (wei2912) [inline assets/js/main.js:281]: Could we use [template literals](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for the lines here?

File: assets/js/main.js [hunk 2]
```diff
@@ -266,95 +265,43 @@ function modal() {
...
+            var output = '';
+            data.results.forEach(function (post) {
+                output +=
+                    '<div class="search-result-row">' +
+                    '<a class="search-result-row-link" href="' +
+                    post['url_path'].raw +
+                    '">' +
+                    post.title.raw +
+                    '</a>' +
+                    '</div>';
```
Comment (wei2912) [inline assets/js/main.js:301]: Same applies for this too.

File: assets/js/main.js [hunk 3]
```diff
@@ -266,95 +265,43 @@ function modal() {
     });
 }
 
+function elasticSearch(query, callback) {
+    var base_url = 'https://advisorysg.ent.ap-southeast-1.aws.found.io';
+    var engine = 'ghost';
+    var search_only_key = 'search-brya3eig6n5g9ybimkw3o9u3';
+
+    var oReq = new XMLHttpRequest();
```
Comment (wei2912) [inline assets/js/main.js:273]: What does `oReq` stand for?

File: assets/js/main.js [hunk 4]
```diff
@@ -266,95 +265,43 @@ function modal() {
     });
 }
 
+function elasticSearch(query, callback) {
+    var base_url = 'https://advisorysg.ent.ap-southeast-1.aws.found.io';
+    var engine = 'ghost';
+    var search_only_key = 'search-brya3eig6n5g9ybimkw3o9u3';
```
Comment (wei2912) [inline assets/js/main.js:271]: Think we should stick to `camelCase` for variable names.

**Review by wei2912 — CHANGES_REQUESTED** [review]
LGTM except for some minor code style changes. Thanks for the PR!

File: assets/js/main.js (hunk 3 above)
Comment (Hackin7) [reply assets/js/main.js]: Changed it to a more meaningful name, searchReq

**Review by wei2912 — APPROVED** [review]
LGTM!

**Pull request merged**